server/src/main/scala/net/wiringbits/actions/SendVerifyEmailAction.scala

```diff
@@ -0,0 +1,68 @@
+package net.wiringbits.actions
+
+import net.wiringbits.apis.models.EmailRequest
+import net.wiringbits.apis.{EmailApi}
+import net.wiringbits.config.{UserTokensConfig, WebAppConfig}
+import net.wiringbits.repositories.UsersRepository
+import net.wiringbits.util.{EmailMessage, TokenGenerator, TokensHelper}
+import net.wiringbits.validations.{ValidateEmailIsRegistered}
+import net.wiringbits.repositories.models.{UserToken, UserTokenType}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+import net.wiringbits.api.models.SendVerifyEmail
+import net.wiringbits.repositories.UserTokensRepository
+import java.time.temporal.ChronoUnit
+import java.time.{Clock, Instant}
+import java.util.UUID
+
+class SendVerifyEmailAction @Inject() (
+    usersRepository: UsersRepository,
+    tokenGenerator: TokenGenerator,
+    userTokensConfig: UserTokensConfig,
+    userTokensRepository: UserTokensRepository,
+    webAppConfig: WebAppConfig,
+    emailApi: EmailApi
+)(implicit
+    ec: ExecutionContext,
+    clock: Clock
+) {
+
+  def apply(request: SendVerifyEmail.Request): Future[SendVerifyEmail.Response] = {
+    for {
+      _ <- validations(request)
+
+      userMaybe <- usersRepository.find(request.email)
+      user = userMaybe.getOrElse(throw new RuntimeException(s"User with email ${request.email} wasn't found"))
+
+      token = tokenGenerator.next()
+      hmacToken = TokensHelper.doHMACSHA1(token.toString.getBytes(), userTokensConfig.hmacSecret)
+      userTokens <- userTokensRepository.find(user.id)
```

Comment:
I'd suggest to not send the token if we have an active token created in the last 5 minutes.